### PR TITLE
Points login page

### PR DESCRIPTION
**Pull request by mishramonalisha76**
### Ticket Number

- [1680](#1680 )

### Description

- **Problem/Feature**:
    Login feature for points vault page and TextInput block component

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe):

### Checklist

- [ ] **Quick PR**: Is this a quick PR? Can be approved before finishing a coffee.
  - [ ] Quick PR label added
- [ ] **Not Merge Ready**: Is this PR dependent on some other PR/tasks and not ready to be merged right now.
  - [ ] DO NOT Merge PR label added

### Frontend Guidelines

- [ ] Followed frontend guidelines as per the [Guidelines Notion Doc](https://www.notion.so/pushprotocol/Frontend-dApp-Guidelines-1d7806ae3d9e4569a340b563dcd0536c)

### Build & Testing

- [x] No errors in the build terminal
- [x] Engineer has tested the changes on their local environment
- [ ] Engineer has tested the changes on deploy preview

### Screenshots/Video with Explanation

- **Before:** Explain the previous behavior

- **After:** What's changed now
- 
image
image
image
image
image

### Review & Approvals

- [ ] Self-review completed
- [ ] Code review by at least one other engineer
- [ ] Documentation updates if applicable

### CODE REVIEW

File: src/blocks/textInput/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        required={required}
+        role="input"
+        mode={mode}
+        ref={ref}
+        {...props}
+      >
+        {(label || totalCount) && (
+          <div className="label-count">
+            {label && <Text className="label">{label}</Text>}
+            {totalCount && <Text className="count">{totalCount}</Text>}
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:143]:
```suggestion
            {label && <Text className="label">{label}</Text>}
            {totalCount && <Text className="count">{totalCount}</Text>}
```

For text the styling should not come from className but from the variant prop. If the variant is not available please ask zee to provide it and then we also add it in Text.

File: src/blocks/textInput/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+              height="18px"
+              className="clear"
+              onClick={() => onClear?.()}
+            />
+          )}
+        </div>
+        {description && (
+          <Text
+            variant="c-regular"
+            className="description"
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:169]: Same we will not be providing className to Text

File: src/blocks/textInput/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      onClear,
+      description,
+      errorMessage,
+
+      ...props
+    },
+    ref
+  ) => {
+    const { mode } = useBlocksTheme();
+    const state = getTextInputState({ error: !!error, disabled: !!disabled });
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:129]: This is unused

File: src/blocks/textInput/TextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      totalCount,
+      placeholder,
+      icon,
+      type = 'text',
+      onChange,
+      value,
+      onClear,
+      description,
+      errorMessage,
+
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:123]: Remove spacing

File: src/blocks/textInput/TextInput.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+  disabled?: boolean;
+  /* Description shown below the input field */
+  description?: string;
+  /* Error message shown below the input field */
+  errorMessage?: string;
+};
+
+const StyledTextInput = styled.div<TextInputProps & ModeProp>`
+  /* Common Input field CSS */
+  display: flex;
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:47]: Format the complete css

File: src/blocks/textInput/TextInput.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+      span {
+        width: 18px;
+        height: 13.5px;
+      }
+    }
+
+    & input {
+      border: none;
+      background-color: transparent;
+      width: 100%;
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.tsx:85]: The inout width should be auto or inherit and it should support a width prop to set the custom width. Please check out other design systems on how they work

File: src/blocks/textInput/TextInput.utils.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  return 'default';
+};
+
+export const getTextInputStateStyles = ({ mode, state }: { mode: ThemeMode; state: TextInputStates }) => {
+  /*check all dark mode , label ,count and icon and placeholder colors
+  add success state
+  */
+  return `
+
```
Comment (rohitmalhotra1420) [inline src/blocks/textInput/TextInput.utils.ts:21]: formatting if off for css

File: src/modules/pointsVault/PointsVault.tsx [hunk 8]
```diff
@@ -2,18 +2,17 @@ import { Box } from 'blocks';
...
 import { PointsVaultLogin } from './components/PointsVaultLogin';
 import { PointsVaultList } from './components/PointsVaultList';
+import { PointsVaultComponents } from './PointsVault.types';
 
-export type PointsVaultProps = {};
-
-const PointsVault: FC<PointsVaultProps> = () => {
-  const [activeComponent, setActiveComponent] = useState('login');
+const PointsVault: FC = () => {
+  const [activeComponent, setActiveComponent] = useState<PointsVaultComponents>('login');
```
Comment (rohitmalhotra1420) [inline src/modules/pointsVault/PointsVault.tsx:8]:
```suggestion
  const [activeView, setActiveView] = useState<PointsVaultView>('login');
```

File: src/queries/hooks/pointsVault/usePointsVaultUserLogin.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { useMutation } from '@tanstack/react-query';
+
+import { vaultUserLogin } from '../../queryKeys';
+
+import { PointsVaultUserLoginProps } from '../../types';
+import { pointsVaultUserLogin } from '../../services';
+
+export const usePointsVaultUserLogin = (payload: PointsVaultUserLoginProps) =>
```
Comment (rohitmalhotra1420) [inline src/queries/hooks/pointsVault/usePointsVaultUserLogin.ts:8]:
```suggestion
export const usePointsVaultUserLogin = () =>
```

File: src/modules/pointsVault/components/PointsVaultLogin.tsx [hunk 10]
```diff
@@ -1,10 +1,151 @@
...
+
+export type PointsVaultLoginProps = {
+  handleSetActiveComponent: (component: PointsVaultComponents) => void;
+};
+
+const PointsVaultLogin: FC<PointsVaultLoginProps> = ({ handleSetActiveComponent }) => {
+  const { mutate: pointsVaultUserLogin, isPending } = usePointsVaultUserLogin({
+    username: '',
+    password: '',
+  });
```
Comment (rohitmalhotra1420) [inline src/modules/pointsVault/components/PointsVaultLogin.tsx:23]:
```suggestion
  const { mutate: pointsVaultUserLogin, isPending } = usePointsVaultUserLogin();
```

File: src/modules/pointsVault/components/PointsVaultLogin.tsx [hunk 11]
```diff
@@ -1,10 +1,151 @@
...
+
+  const validationSchema = yup.object().shape({
+    username: yup.string().required('Username is required'),
+    password: yup.string().required('Password is required'),
+  });
+
+  const formik = useFormik({
```
Comment (rohitmalhotra1420) [inline src/modules/pointsVault/components/PointsVaultLogin.tsx:30]:
```suggestion
  const formik = useFormik<VaultLoginformValues>({
```

export type VaultLoginFormValues = { username: string, password:string }

File: src/modules/pointsVault/components/PointsVaultLogin.tsx [hunk 12]
```diff
@@ -1,10 +1,151 @@
...
+      </Box>
+    </Box>
+  );
 };
 
 export { PointsVaultLogin };
+
+//styles
+
+const StyledForm = styled.form`
```
Comment (rohitmalhotra1420) [inline src/modules/pointsVault/components/PointsVaultLogin.tsx:149]: Can't we do this?

<Box width="100%"><form> ... </form></Box>

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

File: package.json [hunk 13]
```diff
@@ -117,7 +118,8 @@
     "styled-components": "^5.3.8",
     "url": "0.11.0",
     "web3": "^1.8.2",
-    "xss": "^1.0.14"
+    "xss": "^1.0.14",
```
[the remark this answers is not in the document]
Comment (mishramonalisha76) [reply package.json:121]: not sure it is used somewhere else, i havent installed it

File: src/blocks/textInput/TextInput.tsx (hunk 1 above)
Comment (mishramonalisha76) [reply src/blocks/textInput/TextInput.tsx:143]: removing the class name but the styling for the same isnt done in this pr , will do it in the later ticket for textInput

File: package.json (hunk 13 above)
[the remark this answers is not in the document]
Comment (mishramonalisha76) [reply package.json:121]: A older component is using it

**Pull request merged**